Ensuring that links point to index meta-dbs (#34)
Materials-Consortia/providers: src/index-metadbs/README.md

# Static Index Meta-Databases

## Goal

> This folder hosts static versions of Index Meta-Databases for those providers that only have one main sub-database (or very few sub-databases) and do not wish to maintain an Index Meta-Database themselves (Index Meta-Databases are described in the [OPTIMADE specifications](https://github.com/Materials-Consortia/OPTIMADE/blob/develop/optimade.rst)).

Reviewer: Do you want to also use the header tag in this link, or?

Author: I removed it on purpose as I wanted to point to v1 once is released, and I feel we'll forget to fix this file (while we will remember the README). So I preferred to keep a generic link.
I can put it back if you want. But we'll need to remember to adapt it once the specs are out.

Author: I see now that in the rest you added them. I'll then do the same.

Reviewer: Yeah, I'll create a preemtive issue stating these links should be updated come the release of v1.0.

> Note: while providing an (empty) Index Meta-Database is not required by the OPTIMADE API specifications, it is instead required in order to be listed in this List of Providers ([providers.optimade.org](http://providers.optimade.org)).

Reviewer: ```suggestion
Note: while providing an Index Meta-Database is not required by the OPTIMADE API specification, it is instead required in order to be listed in this List of Providers ([providers.optimade.org](http://providers.optimade.org)).
```

> ## When should I use a static index meta-database hosted in this repository

Reviewer: Just to keep the same way of writing that you have used throughout otherwise.
```suggestion
## When should I use a static Index Meta-Database hosted in this repository
```

If you add a new provider and do not wish to host an Index Meta-Database yourself, feel free to add an Index Meta-Database here.

> However, if your Index Meta-Database is rapidly changing (e.g. because the list of sub-databases changes often), please host it yourself to avoid multiple commits and pull requests to this repository at every change.

Reviewer: ```suggestion
However, if your Index Meta-Database is rapidly changing (e.g., because the list of sub-databases changes often), please host it yourself to avoid multiple commits and pull requests to this repository at every change.
```

Rare changes are instead of course welcome, e.g., if you have to change your DNS name or the description, or if you need just to add one static new sub-database, like a test database.

Note that "changes" here refer solely to changes to the *list of sub-databases*; the content of each sub-database can change at any time without the need to modify the Index Meta-Database.

## Instructions on how to add a new static Index Meta-Database

> 1. Go in the folder `src/index-metadbs`, copy one of the existing folders (e.g. `cod`) to a new folder, where the folder name should be the identifier of your provider (we will use `exmpl` here and in the following).

Reviewer 2: I think it is a bit inelegant to tie this document to an actual specific provider (cod). Can you instead create a 'template' directory that people can copy?

Author: Sure

Author: Done in f27908a9643b9259a53b9cec0b468fef60f70829

> 1. Go in the folder `src/index-metadbs`, copy the existing template folder (`exmpl`) to a new folder, where the folder name should be the identifier of your provider (we will use `exmpl2` here and in the following as the name of the new provider).

Reviewer: ```suggestion

1. Go to the folder `src/index-metadbs`, copy the existing template folder (`exmpl`) to a new folder, where the new folder name should be the identifier of your provider (we will use `exmpl2` here and in the following as the name of the new provider).
```

   In particular, this will contain a `v1` subfolder, with two files inside it: `info.json` and `links.json`.

> 2. Adapt the content of the `links.json` file. In particular, provide a new `link` dictionary entry for each sub-database that you want to refer to. 

Reviewer: ```suggestion
2. Adapt the content of the `links.json` file.
   In particular, provide a new `links` entry resource object for each sub-database that you want to link to.
```

   In the special case in which you have a single "main" sub-database, just change the existing values as follows:

   - change the `id` to your provider identifier: `"exmpl" -> "exmpl2"`

>    - change the attributes `name`, `description` and `homepage` to contain the correct content. Please reuse the *same* content as the one you specified in the main `providers.json` file.

Reviewer: ```suggestion
   - change the attributes `name`, `description` and `homepage` to contain the correct content.
     Please reuse the *same* content as the one you specified in the main `providers.json` file.
```

>    - point the `base_url` to the base URL of your OPTIMADE implementation.

Reviewer: ```suggestion
   - point the `base_url` to the base URL of your OPTIMADE implementation.
     Note, this should be the un-versioned base URL (see [the OPTIMADE API specification](https://github.com/Materials-Consortia/OPTiMaDe/blob/develop/optimade.rst#base-url) for more information on this).
```

> 3. Adapt the content of the `info.json` file. In particular, you should change two fields:

Reviewer: ```suggestion
3. Adapt the content of the `info.json` file.
   In particular, you should change two fields:
```

   - change the URL of the `available_api_versions` by replacing `exmpl` with your identifier: `http://providers.optimade.org/index-metadbs/exmpl2/v1/`

>    - change the `id` inside `data->relationships->default->data->id` from `exmpl` to the correct ID from the list of links in the `links.json` file.

Reviewer: ```suggestion
   - change the `id` inside `data -> relationships -> default -> data -> id` from `exmpl` to the correct ID from the list of links in the `links.json` file.
```

>      As explained in the OPTIMADE specifications, this should be the ID of the database that should be considered as the "default" sub-database by clients. 

Reviewer: ```suggestion
     As [explained in the OPTIMADE API specification](https://github.com/Materials-Consortia/OPTiMaDe/blob/develop/optimade.rst#base-info-endpoint), this should be the ID of the database that should be considered as the "default" sub-database by clients.
```

     
     If you only have one sub-database and you followed the instructions above, you should use here your provider identifier.

>      If you do not wish to have a default database, remove completely the `relationships` attribute.

Reviewer: I'm not sure this is what they should do, but rather it should be "unknown", i.e., `null`.
```suggestion
     If you do not wish to have a default database, set the `relationships` value to `null`.
```
